src/klipse/plugin.cljs

```diff
@@ -105,24 +107,46 @@
   [element general-settings mode]
   (if-let [opts (@mode-options mode)]
     ;; weird piece of code - see klipsify-with-opts docstring
-    (go (<! ((<! (klipsify-with-opts element general-settings opts mode)))))
+    (go (<! ((<! (klipsify-with-opts element general-settings opts mode #())))))
     (go (js/console.error "cannot find options for mode: " mode ". Supported modes: " (keys @mode-options)))))
 
-(defn ^:export klipsify-no-eval [element general-settings mode]
+(defn ^:export klipsify-no-eval [element general-settings mode on-edit-cp]
   (if-let [opts (@mode-options mode)]
-    (klipsify-with-opts element general-settings opts mode)
+    (klipsify-with-opts element general-settings opts mode on-edit-cp)
     (go #(go (js/console.error "cannot find options for mode: " mode ". Supported modes: " (keys @mode-options))))))
 
 (defn edit-elements [elements general-settings modes]
   (go
-    (loop [elements elements eval-fns []]
-      (if (seq elements)
-        (let [element (first elements)]
-          (if-let [mode (modes element)]
-            (let [eval-fn (<! (klipsify-no-eval element general-settings mode))]
-              (recur (rest elements) (conj eval-fns eval-fn)))
-            (recur (rest elements) eval-fns)))
-        eval-fns))))
+    (let [valid-elements (filter modes elements)
+          ; these channels are posted to whenever an edit is made
+          event-chans (mapv (fn [_] (chan (if (general-settings :re_evaluate_all_snippets_on_change)
+                                            10
```

Comment:
this code looks very smart.
Could you make it easier to read (maybe split into functions) and also explain me what the code does?

Reply:
@viebel This code creates a channel for each snippet. Whenever a snippet is edited, it's index is pushed on the channel. And the subprocess reloads all snippets except that since it has already been reloaded.

As for refactoring, I can not see how this code can be split into functions. Any pointers?

Reply:
Would it make the code simpler if we evaluate the edited snippet again?
I want to keep the reevaluation logic as simple as possible: whenever a change is made to a snippet, we re-evaluate all the snippets - including the edited snippet - in the order they appear on the page. 
Later, we might support snippet dependency.

It's not what is suggested in #361 but I think it's preferable. What do you think?